src/paths/all_shortest_paths.c

```diff
@@ -175,6 +188,10 @@ int igraph_get_all_shortest_paths(const igraph_t *graph,
     IGRAPH_CHECK(igraph_vector_ptr_push_back(&paths, vptr));
     IGRAPH_CHECK(igraph_vector_init(vptr, 1));
     VECTOR(*vptr)[0] = from;
+    vptr_e = igraph_Calloc(1, igraph_vector_t);
+    IGRAPH_CHECK(igraph_vector_ptr_push_back(&path_edge, vptr_e));
+    IGRAPH_CHECK(igraph_vector_init(vptr_e, 1));
+    VECTOR(*vptr_e)[0] = -1;
```

Comment:
Not sure if I like this -1 in the edge ID vector; couldn't we simply keep the vector empty?

Reply:
well I prefer not, because the vector `vptr_e` is initialized to 1.
Is it O.K?

Reply:
But why can't we simply initialize `vptr_e` with zero elements? (i.e. `igraph_vector_init(vptr_e, 0)`.